source/client-side-encryption/client-side-encryption.md

```diff
@@ -472,6 +473,34 @@ See
 <span id="GCPKMSOptions"></span> <span id="AWSKMSOptions"></span> <span id="KMSProvider"></span>
 <span id="KMSProviders"></span> <span id="AzureAccessToken"></span> <span id="kmsproviders"></span>
 
+#### credentialProviders
+
+The `credentialProviders` property may be specified on [ClientEncryptionOpts](#ClientEncryptionOpts) or
```

Comment:
Do you think it would be helpful to throw errors if the user supplies a credentialProvider configuration but their kmsProviders are not configured for automatic credential fetching?

```typescript
new ClientEncryption(client, { 
  kmsProviders: { aws: { accessKeyId: .. } },
  credentialProviders: { aws: .. }
});

// MongoCryptError: credentialProviders can only be used with automatic KMS credential fetching.
```

Reply:
Good idea. I've updated to indicate an error must be thrown in this situation.